src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/StackGuard.cs

```diff
@@ -62,12 +62,12 @@ public bool TryEnterOnCurrentStack()
         {
             return RunOnEmptyStackCore(s =>
             {
-                var t = (Tuple<Func<T1, T2, T3, R>, T1, T2, T3>)s;
+                var t = (Tuple<Func<T1, T2, T3, R>, T1, T2, T3>)s!;
                 return t.Item1(t.Item2, t.Item3, t.Item4);
             }, Tuple.Create(action, arg1, arg2, arg3));
         }
 
-        private R RunOnEmptyStackCore<R>(Func<object, R> action, object state)
+        private R RunOnEmptyStackCore<R>(Func<object?, R> action, object state)
```

Comment:
>object? [](start = 46, length = 7)

Can this be `object` rather than `object?`? Would that remove the need for `!` in callers above?

Reply:
Yes that would remove need for `!` in callers, but would need to add `!` for ` Task.Factory.StartNew(action!, ...)` below, as its need only one `!` will update